src/api/app/models/user.rb

```diff
@@ -903,6 +900,16 @@ def update_notifications(params)
 
   private
 
+  def involved_packages_ids
+    user_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id) +
+        group_maintained_packages.where.not(project_id: involved_projects_ids).pluck(:id)
```

Comment:
After thinking about it:

``
Package.joins(:relationships).where(relationships: { user_id: id }).or(
        Package.joins(:relationships).where(relationships: { group_id: groups.pluck(:id) })
    )
``

What do you think about it?